src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -931,19 +925,35 @@ abstract class ManagedStrategy(
             val count = currentThreadCodeLocationVisitCountMap.getOrDefault(codeLocation, 0) + 1
             currentThreadCodeLocationVisitCountMap[codeLocation] = count
             currentThreadCodeLocationsHistory += codeLocation
+            val detectedFirstTime = count > hangingDetectionThreshold
             val detectedEarly = loopTrackingCursor.isInCycle
             // Check whether the count exceeds the maximum number of repetitions for loop/hang detection.
-            val detectedFirstTime = count > hangingDetectionThreshold
             if (detectedFirstTime && !detectedEarly) {
```

Comment:
No, it can happen sometimes when we can't find a cycle period and can't switch to another thread. An example is `org.jetbrains.kotlinx.lincheck_test.representation.ObstructionFreedomActiveLockRepresentationTest`

Reply:
Please add a comment about all the possible cases.